tests/test_tools.py

```diff
@@ -2174,5 +2179,105 @@ def test_get_tags_invalid():
         get_tags(coll)
         assert e_info == 'ERROR: valid tags are comma or space separated strings of tag names'
 
+@pytest.mark.parametrize(
+    "repo_information, expected", [
+        ({"repos": {
+            "repo1": {
+                "params": {
+                    "namespace_id": "35",
+                    "initialize_with_readme": "false",
+                    "name": "repo name"},
+                "url": "https://gitlab.thebillingegroup.com/api/v4/projects/"}}
+         }, True),
+        ({}, False),
+        ({"repos": {}}, False),
+        ({"repos": {
+            "repo1": {}, }
+         }, False),
+        ({"repos": {
+            "repo1": {
+                "url": "https://gitlab.thebillingegroup.com/api/v4/projects/"}}
+         }, False),
+        ({"repos": {
+            "repo1": {
+                "params": {},
+                "url": "https://gitlab.thebillingegroup.com/api/v4/projects/"}}
+         }, False),
+        ({"repos": {
+            "repo1": {
+                "params": {
+                    "namespace_id": "",
+                    "initialize_with_readme": "false",
+                    "name": "repo name"},
+                "url": "https://gitlab.thebillingegroup.com/api/v4/projects/"}}
+         }, False),
+        ({"repos": {
+            "repo1": {
+                "params": {
+                    "namespace_id": "35",
+                    "initialize_with_readme": "false",
+                    "name": "repo name"}}}
+         }, False),
+        ({"repos": {
+            "repo1": {
+                "params": {
+                    "namespace_id": "35",
+                    "initialize_with_readme": "false",
+                    "name": "repo name"},
+                "url": ""}}
+         }, False),
+        ({"repos": {
+            "repo1": {
+                "params": {
+                    "namespace_id": "35",
+                    "initialize_with_readme": "false",
+                    "name": "repo name"},
+                "url": "https://example.com"}}
+         }, False)
+    ]
+)
+def test_repo_info_complete(repo_information, expected):
```

Comment:
is this testing all the functionality?  for example that it is raising the right exception and giving the right error message in different situations?

Reply:
Yes, I just reviewed all the test cases and cross checked them with everything I wanted to test for in the functionality. I made some tweaks to the code in the process while refining the tests as well